tests/functional/test_merging_data_sources.py

```diff
@@ -0,0 +1,80 @@
+import pytest
+
+
+from application.auth.models import TypeOfUser
+from tests.functional.utils import driver_login
+from tests.models import DataSourceFactory, UserFactory
+from selenium.common.exceptions import NoSuchElementException
+
+
+def test_merging_two_data_sources(driver, live_server):
+
+    # Setup
+    DataSourceFactory.create(title="Police statistics 2019")
+    DataSourceFactory.create(title="Police stats 2019")
+    admin_user = UserFactory(user_type=TypeOfUser.ADMIN_USER, active=True)
+
+    driver_login(driver, live_server, admin_user)
+
+    # When I go to the admin page
+    driver.get(f"http://localhost:{live_server.port}/admin")
```

Comment:
I don't have strong opinions here but this isn't consistent with how we do our other functional tests. For each page, we generally have a class that we use to navigate/interact with the page.

Something like:

```
class AdminIndexPage(BasePage):
    def __init__(self, driver, live_server):
        super().__init__(driver=driver, base_url="http://localhost:%s/admin" % live_server.port)
```

Reply:
It's definitely inconsistent. But personally I find the levels of abstraction in the other functional tests not that helpful, and fairly confusing. eg [test_secondary_sources.py](https://github.com/racedisparityaudit/ethnicity-facts-and-figures-publisher/blob/026849a19911f180ccb6b40fde335d4dc52a7f99/tests/functional/test_secondary_sources.py) doesn’t actually contain any assertions.

That said, I'll admit I did it this way partly because it was way quicker to write, and as a bit of an experiment in sticking purely to the Selenium api.

Reply:
I'm not against changing, but I tend to think that being consistent is better than having little pockets of different systems/frameworks.

I honestly thought you were a proponent of the page-model functional testing thing. A little surprised to hear you aren't a fan.